### ISSUE TITLE
Step does change versionCode but not versionName

### ISSUE THREAD

**Issue by tHollarek** [bitrise-community/steps-change-android-versioncode-and-versionname#9]
Hey everyone,

I think my problem is somewhat similar to the other open one at the moment.

I have configured the step to change the versionCode and the versionName of my app during a deploy workflow.

The name and code are created from variables that I set correctly.
However the script then changes the versionCode but not the versionName and continues without an error as well.

Here is the log output:

Configs:
- BuildGradlePth: ./app/build.gradle
- NewVersionName: 1.3.0 (12)
- NewVersionCode: 12
- VersionCodeOffset: 
Updating versionName and versionCode in: ./app/build.gradle
updating line (20):         versionCode 11 ->         versionCode 12
1 versionCode updated
0 versionName updated

**Comment by trapacska** [comment]
Hey @tHollarek !

Thank you for reporting the issue. Please attach your corresponding build.gradle file (Of course with censored sensitive data values, but please try to keep the original look of the gradle file.), and then I can schedule it in out internal issue tracker to fix. Thank you!

**Comment by tHollarek** [comment]
Hey,

here is the censored gradle file. Let me know if you need any additional info to look into this.

Thanks for taking care! :)

lg,
Tobi
```

apply plugin: 'com.android.application'
apply plugin: 'realm-android'
apply plugin: 'io.fabric'
android {
   compileSdkVersion rootProject.sdkVersion
   buildToolsVersion rootProject.buildTools


    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId 'de.package.name'
        targetSdkVersion rootProject.sdkVersion
        minSdkVersion rootProject.minSdkVersion

        multiDexEnabled true
        versionCode 9
        versionName '0.0.9'

        vectorDrawables.useSupportLibrary = true
        testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'

        //render Blur
        renderscriptTargetApi 22
        renderscriptSupportModeEnabled true

        //vector support
        vectorDrawables.useSupportLibrary = true
    }

    dexOptions {
        javaMaxHeapSize '4g'
    }

    buildTypes {
        debug {
            applicationIdSuffix '.debug'
        }
        staging {
            applicationIdSuffix '.staging'
            signingConfig signingConfigs.debug
            debuggable true
        }
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }

    testOptions {
        unitTests.returnDefaultValues = true
    }
}

dependencies {
    implementation some.libraries
    androidTestImplementation some.libraries
    testImplementation some.libraries
}

apply plugin: 'com.google.gms.google-services'
```

**Comment by BirmacherAkos** [comment]
Hi!

It seems the step has an issue with the versionName regexp.
It will search for:
`versionName "0.0.9"`

And you have:
`versionName '0.0.9'`

The step will find the versionCode but the regex will fail at versionName (because of the `'` character).
Because of that, it will find the versionCode only.

And because the step doesn't need both of the versionName and versionCode it will not fail.

Sorry for the inconvenience.
I have made a ticket for that issue.

**Comment by tHollarek** [comment]
Hey,

thanks! :) any estimation as to when that will be released?

I can probably fix the issue for now by just changing my gradle file to use the double quotes, right?

**Comment by BirmacherAkos** [comment]
Hi @tHollarek!

We have released a new version [(1.1.1)](https://github.com/bitrise-community/steps-change-android-versioncode-and-versionname/releases/tag/1.1.1) Change Android versionCode and versionName Step.
It should fix the issue.

Happy Building! :)
